### PR TITLE
Redesign Docker Image and Github Actions

### PR DESCRIPTION
**Pull request by Synicix**
- Redesign the docker image to take build args for different pytorch and cuda version
- Rewrote the github actions to take arguments and different configurations of cuda and pytorch
- Combined github actions into one yaml file with docker login and uploading image restricted to main fork and branch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+ARG UBUNTU_VER
+ARG CUDA_VER
+
+FROM nvidia/cuda:${CUDA_VER}-devel-ubuntu${UBUNTU_VER}
+ARG TORCH_VER
+ARG TORCHVISION_VER
+ARG TORCHAUDIO_VER
+ARG PYTORCH_EXTRA_IDX_URL
+
 LABEL maintainer="Edgar Y. Walker <eywalker@uw.edu>"
```
Comment (eywalker) [inline Dockerfile]: Could you also add yourself to the maintainer @Synicix ?

File: Dockerfile [hunk 2]
```diff
@@ -38,35 +47,8 @@ RUN apt-get update && \
     # best practice to keep the Docker image lean
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
-# Install essential Python packages
-RUN pip3 --no-cache-dir install \
-    numpy \
-    matplotlib \
```
Comment (eywalker) [inline Dockerfile:44]: So in this Docker image, we are not going to put any additional scientific packages, are we? I guess we'll put it in a downstream image?

**Review by eywalker — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (Synicix) [reply Dockerfile:44]: Yep, going to move everything down stream for that

File: Dockerfile (hunk 1 above)
Comment (Synicix) [reply Dockerfile]: Sure

**Review by eywalker — APPROVED** [review]

**Pull request merged**